client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,634 @@
+package zio.raft.client
+
+import zio._
+import zio.stream._
+import zio.raft.protocol._
+import scodec.bits.ByteVector
+import java.time.Instant
+
+/**
+ * Main RaftClient implementation using functional state-based approach.
+ * 
+ * Each connection state (Disconnected, Connecting, Connected) handles messages
+ * differently and transitions to other states based on the protocol.
+ * 
+ * Session management is integrated directly into the state machine.
+ */
+class RaftClient private (
+  stateRef: Ref[ClientState],
+  pendingOperationsRef: Ref[Map[Nonce, SessionOperation]],
+  serverRequestTrackerRef: Ref[ServerRequestTracker],
+  retryManager: RetryManager,
+  zmqTransport: ZmqClientTransport,
+  config: ClientConfig
+) {
+  
+  /**
+   * Connect to the cluster and create a session.
+   */
+  def connect(): Task[SessionId] = 
+    for {
+      // Validate capabilities
+      _ <- validateCapabilities(config.capabilities)
+      
+      // Transition to Connecting state
+      _ <- transitionTo(ClientState.Connecting(config.capabilities))
+      
+      // Generate nonce for session creation
+      nonce <- Nonce.generate()
+      
+      // Create promise for session creation
+      promise <- Promise.make[SessionError, SessionId]
+      
+      // Track pending operation
+      _ <- pendingOperationsRef.update(_.updated(nonce, SessionOperation.Create(promise)))
+      
+      // Send CreateSession message
+      createMessage = CreateSession(config.capabilities, nonce)
+      _ <- zmqTransport.sendMessage(createMessage)
+      
+      // Wait for session creation response
+      sessionId <- promise.await
+      
+      _ <- ZIO.logInfo(s"Connected with session: $sessionId")
+      
+    } yield sessionId
+  
+  /**
+   * Continue an existing session after reconnection.
+   */
+  def continueSession(sessionId: SessionId): Task[Unit] = 
```

Comment:
we don't need this method at all. Continue Session happens automatically whenever we are trying to reconnect, but we already have a session id. No one should ever call this.